hack/validation/labels.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# labels validation for nodepool
+# checking for restricted labels while filtering out well known labels
+
+rule=$'self.all(x, x in
+    [
+        "karpenter.azure.com/sku-name",
+        "karpenter.azure.com/sku-family",
+        "karpenter.azure.com/sku-version",
+        "karpenter.azure.com/sku-cpu",
+        "karpenter.azure.com/sku-memory",
+        "karpenter.azure.com/sku-accelerator",
+        "karpenter.azure.com/sku-networking-accelerated",
+        "karpenter.azure.com/sku-storage-premium-capable",
+        "karpenter.azure.com/sku-storage-ephemeralos-maxsize",
+        "karpenter.azure.com/sku-encryptionathost-capable",
+        "karpenter.azure.com/sku-gpu-name",
+        "karpenter.azure.com/sku-gpu-manufacturer",
+        "karpenter.azure.com/sku-gpu-count"
+    ]
+    || !x.find("^([^/]+)").endsWith("karpenter.azure.com")
+)
```

Comment:
rule is the same for both of these right? can they just be merged into one script?

Reply:
It is close, but not quite the same, one operates on collection (of labels), another operates on a single thing (key). Yes, there are opportunities for sharing, that's what I mean by improving infra, but readability and result are more important for now.